chainer/functions/activation/relu.py

```diff
@@ -24,36 +24,110 @@ def check_type_forward(self, in_types):
         )
 
     def forward_cpu(self, x):
-        self.retain_inputs(())
         self.retain_outputs((0,))
+        self._use_cudnn = False
```

Comment:
I think it's simpler to make `_use_cudnn = False` a class attribute and only sets explicitly on cuDNN path.
I mean:
```py
class ReLU(function_node.FunctionNode):

    _use_cudnn = False

    ...

    def forward_gpu(self, x):
        if ...:
            self._use_cudnn = True
            ...
        ...
```